pkg/gatewayserver/io/mqtt/mqtt_test.go

```diff
@@ -112,6 +112,8 @@ func TestAuthentication(t *testing.T) {
 			client := mqtt.NewClient(clientOpts)
 			token := client.Connect()
 			if tc.OK {
+				for i := 0; i < 3 && token.WaitTimeout(timeout); i++ {
+				}
```

Comment:
This looks wrong; if you need up to 3 times the wait timeout, why not wait for `3 * timeout` ?

Reply:
Went with this because you don't always have to wait for 3 times the timeout. That would make the tests take an unnecessarily long time under normal circumstances. Thus, we only wait for as along there is a reason to. If the test takes a bit longer, then it will not fail as it currently does.

Reply:
How is this different then?

The `WaitTimeout()` does not actually wait for `timeout`, it waits as long as it's necessary until the wait times out

Reply:
Unless I have misinterpreted the way `WaitTimeout` works:

The loop repeats as long as `WaitTimeout()` returns false (which means that no traffic has arrived yet. For the test we expect it to return `true`. Initially, the test will fail if that does not happen within the initial `timeout` duration. With this loop, we retry a few times (in `timeout` intervals) before we conclude that the test has failed. If `WaitTimeout()` returns `true` fast, then we also finish fast. If it is late, then we give it more time to succeed before failing.

Also doing the same below.

Reply:
Yeah, upon further inspection, I had misinterpreted how `WaitTimeout` worked. Reverted this and doubled the timeout duration instead. I think we will keep getting CI failures even after that, though.